app/services/integration_customers/create_service.rb

```diff
@@ -49,6 +51,12 @@ def link_customer!
 
       if integration&.type&.to_s == 'Integrations::NetsuiteIntegration'
         new_integration_customer.subsidiary_id = subsidiary_id
+        new_integration_customer.save!
```

Comment:
before we didn't save `subsidiary_id`?

Reply:
It seems we were saving only when syncing the customer with the sync checkbox checked. When linking to an existing Netsuite customer we were not saving the subsidiary.